### DISCUSSION TITLE
Own post-processor for JAKAmini

### DISCUSSION THREAD

**Discussion by Ann-pr** [visose/Robots#303] [Q&A] [answered]
Hi everyone, 

I am currently trying to add our JAKAmini Cobot to the robots plugin. I can simulate the robot via the Rhino and XML file. To export the correct robot program I have to change the post professor as well, because the current program has the wrong data.  How can I do this? 
I have received a post processor from the manufacturer that I would like to import it, or  can I rewrite an existing post processor? 
I am happy about any help. 

![Bildschirmfoto 2024-12-11 um 15 35 19](https://github.com/user-attachments/assets/fb0e8a0f-065c-44d9-8a8e-f28f74686378)

Anna

**Comment by visose** [comment, marked as answer]
Hi, supporting a new robot manufacturer requires changes to the project itself. You can fork this repository and add those changes.

But as a workaround, if your robot is very similar to one of the existing supported ones, you might be able to get away with defining it in the XML as in this other brand and you can replace the postprocessor with a custom one for your particular robot through an input in the Load RobotSystem component. This will not require you to build a custom version of this plugin.

You will have to write this postprocessor yourself in C#, VB.NET or IronPython. The existing ones are here: https://github.com/visose/Robots/tree/master/src/Robots/PostProcessors

You just need to create a class in a scripting component that implements the `IPostProcessor` interface. You will have to reference the Robots.dll assembly in the scripting component.

**Reply by uwewoessner** [reply]
Dear visose,
I gave a shot at it and created a SystenJAKA and JKSPostProcessor.
It tested it with the JAKA Mini Cobo but other JAKA Robots should work as well.
See the following pull request: https://github.com/visose/Robots/pull/311
Regards,
 Uwe
@Anna: Benjamin has a compiled version for you to try it out.
